gcp/src/main/java/org/apache/iceberg/gcp/biglake/BigLakeClient.java

```diff
@@ -0,0 +1,304 @@
+import com.google.cloud.bigquery.biglake.v1.DeleteCatalogRequest;
+import com.google.cloud.bigquery.biglake.v1.DeleteDatabaseRequest;
+import com.google.cloud.bigquery.biglake.v1.DeleteTableRequest;
+import com.google.cloud.bigquery.biglake.v1.GetCatalogRequest;
+import com.google.cloud.bigquery.biglake.v1.GetDatabaseRequest;
+import com.google.cloud.bigquery.biglake.v1.GetTableRequest;
+import com.google.cloud.bigquery.biglake.v1.ListDatabasesRequest;
+import com.google.cloud.bigquery.biglake.v1.ListTablesRequest;
+import com.google.cloud.bigquery.biglake.v1.LocationName;
+import com.google.cloud.bigquery.biglake.v1.MetastoreServiceClient;
+import com.google.cloud.bigquery.biglake.v1.MetastoreServiceSettings;
+import com.google.cloud.bigquery.biglake.v1.RenameTableRequest;
+import com.google.cloud.bigquery.biglake.v1.Table;
+import com.google.cloud.bigquery.biglake.v1.TableName;
+import com.google.cloud.bigquery.biglake.v1.UpdateDatabaseRequest;
+import com.google.cloud.bigquery.biglake.v1.UpdateTableRequest;
+import com.google.protobuf.Empty;
+import com.google.protobuf.FieldMask;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Map;
+import java.util.function.Supplier;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.NoSuchNamespaceException;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.exceptions.NotAuthorizedException;
+
+/** A client of Google BigLake service. */
+final class BigLakeClient implements Closeable {
+
+  private final MetastoreServiceClient stub;
+
+  /**
+   * Constructs a client of Google BigLake Service.
+   *
+   * @param settings BigLake service settings
+   */
+  BigLakeClient(MetastoreServiceSettings settings) throws IOException {
+    this.stub = MetastoreServiceClient.create(settings);
+  }
+
+  /**
+   * Constructs a client of Google BigLake Service.
+   *
+   * @param biglakeEndpoint BigLake service gRPC endpoint, e.g., "biglake.googleapis.com:443"
+   */
+  BigLakeClient(String biglakeEndpoint) throws IOException {
+    this(MetastoreServiceSettings.newBuilder().setEndpoint(biglakeEndpoint).build());
+  }
+
+  public Catalog createCatalog(CatalogName name, Catalog catalog) {
+    return convertException(
+        () ->
+            stub.createCatalog(
+                CreateCatalogRequest.newBuilder()
+                    .setParent(LocationName.of(name.getProject(), name.getLocation()).toString())
+                    .setCatalogId(name.getCatalog())
+                    .setCatalog(catalog)
+                    .build()),
+        name.getCatalog());
+  }
+
+  public Catalog getCatalog(CatalogName name) {
+    return convertException(
+        () -> {
+          try {
+            return stub.getCatalog(GetCatalogRequest.newBuilder().setName(name.toString()).build());
+          } catch (PermissionDeniedException e) {
+            throw new NoSuchNamespaceException(
```

Comment:
This looks incorrect to me. If permission was denied, then the database does exist. This should not hide that.

Reply:
We can't tell whether it is permission denied or resource not found from the error. BigLake API returns permission denied for both cases for security considerations.